src/base/Quantity.hh

```diff
@@ -208,16 +221,64 @@ swap(Quantity<U, V>& a, Quantity<U, V>& b) noexcept
  * Convert the given quantity into the native Celeritas unit system.
  *
  * \code
- assert(unit_cast(Quantity<real_type, SpeedOfLight>{1})
-        == 2.998e10 * centimeter/second);
+ assert(from_quantity(Quantity<CLight>{1}) == 2.998e10 * centimeter/second);
  * \endcode
  */
 template<class UnitT, class ValueT>
-CELER_CONSTEXPR_FUNCTION auto unit_cast(Quantity<UnitT, ValueT> quant)
-    -> decltype(ValueT() * UnitT::value())
+CELER_CONSTEXPR_FUNCTION auto from_quantity(Quantity<UnitT, ValueT> quant)
+    -> decltype(auto)
+{
+    return quant.value() * UnitT::value();
+}
+
+//! Old spelling of "from_quantity".
+template<class UnitT, class ValueT>
+[[deprecated("use 'from_quantity'")]] CELER_CONSTEXPR_FUNCTION auto
+unit_cast(Quantity<UnitT, ValueT> quant) -> decltype(auto)
 {
     return quant.value() * UnitT::value();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Create a quantity from a value in the Celeritas unit system.
+ *
+ * This function can be used for defining a constant for use in another unit
+ * system (typically a "natural" unit system for use in physics kernels).
+ *
+ * \code
+ constexpr LightSpeed c = to_quantity<LightSpeed>(constants::c_light);
+ assert(c.value() == 1);
+ * \endcode
+ */
+template<class Q>
+CELER_CONSTEXPR_FUNCTION Q to_quantity(typename Q::value_type value)
+{
+    using value_type = typename Q::value_type;
+    using unit_type  = typename Q::unit_type;
+
+    return Q{value * (value_type{1} / unit_type::value())};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Use the value of a Quantity.
+ *
+ * The redundant unit type in the function signature is to make coupling safer
+ * across different parts of the code and to make the user code more readable.
+ *
+ * \code
+ assert(qvalue<LightSpeed>(LightSpeed{1}) == 1);
+ * \endcode
+ */
+template<class Q, class SrcUnitT, class ValueT>
+CELER_CONSTEXPR_FUNCTION auto qvalue(Quantity<SrcUnitT, ValueT> quant)
```

Comment:
Since the function requires an argument of type `Quantity`, I think we can safely use just `value`